cmd/ctl/pkg/convert/convert.go

```diff
@@ -0,0 +1,254 @@
+			cmdutil.CheckErr(o.Complete(cmd))
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.OutputVersion, "output-version", o.OutputVersion, "Output the formatted object with the given group version (for ex: 'cert-manager.io/v1alpha3').")
+	o.PrintFlags.AddFlags(cmd)
+
+	cmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, "to need to get converted.")
+
+	return cmd
+}
+
+// Complete collects information required to run Convert command from command line.
+func (o *Options) Complete(cmd *cobra.Command) (err error) {
+	err = o.FilenameOptions.RequireFilenameOrKustomize()
+	if err != nil {
+		return err
+	}
+
+	// build the printer
+	o.Printer, err = o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes version command
+func (o *Options) Run() error {
+	builder := new(resource.Builder)
+
+	r := builder.Unstructured().LocalParam(true).ContinueOnError().
+		FilenameParam(false, &o.FilenameOptions).Flatten().Do()
+
+	err := r.Err()
+	if err != nil {
+		return fmt.Errorf("error here: %s", err)
+	}
+
+	singleItemImplied := false
+	infos, err := r.IntoSingleItemImplied(&singleItemImplied).Infos()
+	if err != nil {
+		return fmt.Errorf("error here instead: %s", err)
+	}
+
+	if len(infos) == 0 {
+		return fmt.Errorf("no objects passed to convert")
+	}
+
+	var specifiedOutputVersion schema.GroupVersion
+	if len(o.OutputVersion) > 0 {
+		specifiedOutputVersion, err = schema.ParseGroupVersion(o.OutputVersion)
+		if err != nil {
+			return err
+		}
+	}
+
+	factory := serializer.NewCodecFactory(api.Scheme)
+	serializer := apijson.NewSerializerWithOptions(apijson.DefaultMetaFactory, api.Scheme, api.Scheme, apijson.SerializerOptions{})
+	encoder := factory.WithoutConversion().EncoderForVersion(serializer, nil)
+	objects, err := asVersionedObject(infos, !singleItemImplied, specifiedOutputVersion, encoder)
+	if err != nil {
+		return err
+	}
+
+	return o.Printer.PrintObj(objects, o.Out)
+}
+
+// asVersionedObject converts a list of infos into a single object - either a List containing
+// the objects as children, or if only a single Object is present, as that object. The provided
+// version will be preferred as the conversion target, but the Object's mapping version will be
+// used if that version is not present.
+func asVersionedObject(infos []*resource.Info, forceList bool, specifiedOutputVersion schema.GroupVersion, encoder runtime.Encoder) (runtime.Object, error) {
+	objects, err := asVersionedObjects(infos, specifiedOutputVersion, encoder)
+	if err != nil {
+		return nil, err
+	}
+
+	var object runtime.Object
+	if len(objects) == 1 && !forceList {
+		object = objects[0]
+	} else {
+		object = &metainternalversion.List{Items: objects}
+
+		targetVersions := []schema.GroupVersion{}
+		if !specifiedOutputVersion.Empty() {
+			targetVersions = append(targetVersions, specifiedOutputVersion)
+		}
+		targetVersions = append(targetVersions, schema.GroupVersion{Group: "", Version: "v1"})
```

Comment:
What's this for? To allow converting the List object containing the items?

Reply:
Without this we aren't able to create the v1 kind list object and we'll panic

Reply:
can you add that as a comment?